client/lib/domains/dns/reducer.js

```diff
@@ -46,6 +44,36 @@ function deleteDns( state, domainName, record ) {
 	return update( state, command );
 }
 
+function markDnsForDeletion( state, domainName, record, isBeingDeleted ) {
+	const index = findDns( state, domainName, record ),
```

Comment:
I know it's convenient, but I think it'd be better to use an object to pass extra details (`isBeingDeleted` in this instance), so that when you're calling the function I know that all the parameters mean without checking to the original function's signature.